chain/ethereum/src/chain.rs

```diff
@@ -157,7 +159,15 @@ impl Blockchain for Chain {
             self.eth_adapters.cheapest_with(capabilities)?.clone()
         };
 
-        let ethrpc_metrics = Arc::new(SubgraphEthRpcMetrics::new(self.registry.clone(), &loc.hash));
+        let ethrpc_metrics = self
+            .deployment_rpc_metrics
+            .lock()
+            .unwrap()
+            .entry(String::from(loc.hash.as_str()))
```

Comment:
I'm using an `Arc<Mutex` here to avoid changing `triggers_adapter` to take a `&mut self`. But that could be an alternative.

Reply:
`&mut` on `Chain` instances is unlikely to work. See my suggestion above to not have to track this map in the `Chain`.